php/EE/Model/Base.php

```diff
@@ -0,0 +1,323 @@
+<?php
+
+namespace EE\Model;
+
+use EE;
+
+/**
+ * Base EE Model class.
+ */
+abstract class Base extends \ArrayObject
```

Comment:
Because 
1. It would allow us to access any object property from array notation
2. It allows us to use array_* functions on array of models.
i.e.
```
$sites = Sites::all();
$site_paths = array_column( $sites, 'site_fs_path' );
```

Reply:
I guess you can use ArrayObject https://github.com/EasyEngine/easyengine/pull/1143/files#diff-9a84c75c5309379993ff2ed9362b8772R309 instead of that.
> 	return new static( $arr );

Since you're not going to need other functions of `Base`.

Reply:
Actually the idea is that you'll be able to use both the methods you've defined on Model as well as methods of `ArrayObject`. 

i.e. Its possible to do
```
$sites = Site::all();

$sites[0]->add_default_cron();  //some method defined on site model
$sites[0]->delete();
```

Also if we return an `ArrayObject` we won't be able to update models easily as now like
```
$site = site::find('abc.com');
$site->is_ssl = true;
$site->save(); //If arrayobject is returned we won't be able to do this.
```

Hence It's best to use return Model the model and have the model extend ArrayObject

Reply:
Verified. There's no need to extend `ArrayObject` hence removing it.